lib/helpers/client_schema.js

```diff
@@ -427,6 +434,22 @@ module.exports = function getSchema(provider) {
         }
       });
     }
+
+    processCustomMetadata() {
+      configuration.extraClientMetadata.properties.forEach((prop) => {
+        configuration.extraClientMetadata.validator(prop, this[prop], this);
+      });
+    }
+
+    ensureStripUnrecognized() {
+      const whitelisted = [...RECOGNIZED_METADATA, ...configuration.extraClientMetadata.properties];
+      Object.keys(this).forEach((prop) => {
+        if (prop === 'sector_identifier') return;
```

Comment:
Why `sector_identifier` is a special case?

Reply:
`sector_identifier` is an internal property, not one passed by clients - it is resolved instead.